source/code/plugin/health/health_hierarchy_builder.rb

```diff
@@ -0,0 +1,76 @@
+require 'json'
+module HealthModel
+    class HealthHierarchyBuilder
+
+        attr_accessor :health_model_definition, :monitor_factory
+
+        def initialize(health_model_definition, monitor_factory)
+
+            if !health_model_definition.is_a?(ParentMonitorProvider)
+                raise 'Invalid Type Expected: ParentMonitorProvider Actual: #{@health_model_definition.class.name}'
```

Comment:
So if these will be triggered only by changes in model definition, can you please ensure these are handled at some higher level? Also i am assuming you added few unit tests for this. Also telemetry would be useful here.

Reply:
correct. telemetry, i will sync with you on that

Reply:
Adding the ApplicationInsightsUtility means dependencies on oms_common etc which breaks the unit testability. What I have done is send telemetry at the location where this is being handles (the filter_health_model_builder)